src/cart/crt_init.c

```diff
@@ -523,6 +523,23 @@ prov_settings_apply(bool primary, crt_provider_t prov, crt_init_options_t *opt)
 	if (prov != CRT_PROV_OFI_CXI && prov != CRT_PROV_OFI_TCP)
 		d_setenv("NA_OFI_UNEXPECTED_TAG_MSG", "1", 0);
 
+	/**
+	 * Force specific port range for application when using tcp provider to know what
+	 * ports to open when firewall is used.
+	 */
+	if (!crt_is_service() && (prov == CRT_PROV_OFI_TCP || prov == CRT_PROV_OFI_TCP_RXM)) {
+		uint32_t port_low_range  = UINT32_MAX;
+		uint32_t port_high_range = UINT32_MAX;
+
+		crt_env_get(FI_TCP_PORT_LOW_RANGE, &port_low_range);
+		crt_env_get(FI_TCP_PORT_HIGH_RANGE, &port_high_range);
+
+		if (port_low_range == UINT32_MAX && port_high_range == UINT32_MAX) {
+			d_setenv("FI_TCP_PORT_LOW_RANGE", "20100", 0);
+			d_setenv("FI_TCP_PORT_HIGH_RANGE", "21100", 0);
```

Comment:
why do you need to crt_env_get() high and low beforehand if you are calling d_setenv with overwrite=0 anyway?

Reply:
He is only setting them if they aren't set.  Note the initial setting to UINT32_MAX on 531/532.  This is just setting them to default values.

Reply:
One issue, I think we might have here is with MPI jobs that also use libfabric.   This may be too late to have an effect on ofi.

Reply:
d_setenv("FI_TCP_PORT_LOW_RANGE", "20100", 0);
this accomplishes the same though, sets it only if its not set.
@johannlombardi is the idea to only set them if both are not set vs only 1?

Reply:
Correct, i want to set both of none are set. I am just worried that someone might set low, assuming that there is no high and then I screw it up by setting high myself.